Fixed store switcher doesn't work multistore setup with different product urls issue

janakbhimani opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
Fixed store switcher doesn't work multistore setup with different product urls issue
### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#19714: Fixed store switcher doesn't work multistore setup with different product urls issue

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
Merge pull request #6 from magento/2.3-develop

Update 2.3-develop branch

Commit:
Merge pull request #7 from magento/2.3-develop

Update branch 2.3-develop

Commit:
Fixed store switcher not work multistore setup with different product urls issue

Commit:
Fixed code as per given instrunction in codacy

sreichel commented:
@magento-engcom-team give me test instance

magento-engcom-team commented:
Hi @sreichel. Thank you for your request. I'm working on Magento instance for you

magento-engcom-team commented:
Hi @sreichel, here is your [new Magento instance](https://pr-19798.instances.magento-community.engineering).
Admin access: [https://pr-19798.instances.magento-community.engineering/admin](https://pr-19798.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

sreichel commented:
LGTM

davidverholen commented on app/code/Magento/UrlRewrite/Model/StoreSwitcher/RewriteUrl.php:
```diff
@@ -67,13 +67,18 @@ public function switch(StoreInterface $fromStore, StoreInterface $targetStore, s
+            }
+            foreach ($currentRewrites as $rewrite) {
+                $targetUrl = $targetStore->getBaseUrl() . $rewrite->getRequestPath();
+                break;
```
maybe it's just personal preference, but returning the targetUrl here instead of a break would make it consistent with the if condition before and so maybe make it more readable.

janakbhimani replied:
Hi @davidverholen 
I have updated code as per your instruction, there were no need to break statement, I have removed it. I have checked URL rewrite functionality with multiple store in my local setup, it is working fine. 

Thanks.

davidverholen reviewed (commented):
It seems to be a great addition, did not run into this before. Just one small comment although I'm not sure if it's just personal preference

Commit:
Code updated as per reviever instrunctions

davidverholen commented:
@janakbhimani thx! There is a second pull request basically with the same fix and im currently trying to figure out which one should be merged.

You changed the findOneByData to findAllByData. Do you maybe still remember a usecase where this would be necessary? If it has no impact on functionality I would argue the findOneByData is probably better regarding performance (see https://github.com/magento/magento2/pull/20093 for reference)

Commit:
Fixed Travis CI build failed

Commit:
Fixed travis failed issue

janakbhimani commented:
Hi @davidverholen,
I have updated the code, previously Travis got failed, so I have fixed it, therefore multiple commits are created in this PR, but finally all checks are pass and latest commit is working commit 8a05dbb

Thanks.

magento-engcom-team commented:
Hi @davidverholen, thank you for the review.
[ENGCOM-4050](https://jira.corp.magento.com/browse/ENGCOM-4050) has been created to process this Pull Request

magento-engcom-team commented:
@janakbhimani thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.
